src/screens/AccountNetworkChooser.js

```diff
@@ -63,7 +48,7 @@ class AccountNetworkChooserView extends React.PureComponent {
               }
             ]}
             onPress={() => {
-              accounts.updateNew({ chainId: network.ethereumChainId });
```

Comment:
I am not sure if this part of refactoring does not affect renderings. I generally do not think to pass a function as params in react-navigation is not a good idea, because this will involves `this` pitfall. Could you double-check that, since I think the performance issue is mainly caused by the icon selection part? Another thing is that if we need this part, could we change the function name `onChange` to something like `updateNetwork`, because the name is implicit?

Reply:
the `this` binding problem is handled nicely by the fat arrow (`=>`) function. It works pretty well as far as I could test.

The full story about the performance issue is the following:

You change the `selected.seed` and `selected.address` when selecting an icon. This triggers an `updateNew` that does a bunch of things, one of it being and `await brainWalletAddress(seed);`  (that we actually don't need) and this inside a `debounce( .., 200)`. 

Once this is done, this triggers a re-rendering of the view because the accounts (props) from the store have changed, where `selected = accounts.getNew()` is actualized. Then because `selected,address` is the `value` of `AccountIconChooser` this triggers a re-rendering of it with what was returned by the update. Long story short, if you log `AccountIconChooser` when the icon is selected and what `address` (called value) it receives from the props you'll see. if you click on the 3 or 4 first icons:
```js
selected 1;
selected 2;
selected 3;
get 1;
get 2;
get 3;
```

> Another thing is that if we need this part, could we change the function name onChange to something like updateNetwork, because the name is implicit?

Sure why not

Reply:
Although I agree that passing an onChange function to the navigation params is probably not the best practice, I also think that this component should probably be refactored into a modal or a simple dropdown rather than its own navigation screen with access to the `unstated` accounts.